homeassistant/components/lutron_caseta/__init__.py

```diff
@@ -218,57 +221,158 @@ def _async_register_bridge_device(
+            )
+            keypad["dr_device_id"] = dr_device.id
+            dr_device_id_to_keypad[dr_device.id] = keypad
+
+        # Add button to parent keypad, and build keypad_buttons and keypad_button_names_to_leap
+        button = keypad_buttons[button_device_id] = LutronButton(
+            lutron_device_id=button_device_id,
+            leap_button_number=bridge_button["button_number"],
+            button_name=_get_button_name(keypad, bridge_button),
+            led_device_id=bridge_button.get("button_led"),
+            parent_keypad=keypad["lutron_device_id"],
         )
 
-        if "serial" not in ha_device or ha_device_serial in seen:
-            continue
-        seen.add(ha_device_serial)
-
-        area = _area_name_from_id(bridge.areas, ha_device["area"])
-        # name field is still a combination of area and name from pylytron-caseta
-        # extract the name portion only.
-        name = ha_device["name"].split("_")[-1]
-        device_args: DeviceInfo = {
-            "name": f"{area} {name}",
-            "manufacturer": MANUFACTURER,
-            "identifiers": {(DOMAIN, ha_device_serial)},
-            "model": f"{ha_device['model']} ({ha_device['type']})",
-            "via_device": (DOMAIN, bridge_device["serial"]),
-        }
-        if area != UNASSIGNED_AREA:
-            device_args["suggested_area"] = area
+        keypad["buttons"].append(button["lutron_device_id"])
 
-        dr_device = device_registry.async_get_or_create(
-            **device_args, config_entry_id=config_entry_id
+        keypad_button_names_to_leap.setdefault(keypad["lutron_device_id"], {}).update(
+            {button["button_name"]: int(button["leap_button_number"])}
         )
-        button_devices_by_dr_id[dr_device.id] = ha_device
-        device_info_by_device_id.setdefault(ha_device["device_id"], device_args)
 
-    return button_devices_by_dr_id, device_info_by_device_id
+    keypad_trigger_schemas = _async_build_trigger_schemas(keypad_button_names_to_leap)
+
+    _async_subscribe_keypad_events(hass, bridge, keypads, keypad_buttons)
+
+    return LutronKeypadData(
+        dr_device_id_to_keypad,
+        keypads,
+        keypad_buttons,
+        keypad_button_names_to_leap,
+        keypad_trigger_schemas,
+    )
+
+
+@callback
+def _async_build_trigger_schemas(
+    keypad_button_names_to_leap: dict[int, dict[str, int]]
+) -> dict[int, vol.Schema]:
+    """Build device trigger schemas."""
+
+    keypad_trigger_schemas: dict[int, vol.Schema] = {}
+
+    for keypad_id, _ in keypad_button_names_to_leap.items():
+        keypad_trigger_schemas.setdefault(
+            keypad_id,
+            LUTRON_BUTTON_TRIGGER_SCHEMA.extend(
+                {
+                    vol.Required(CONF_SUBTYPE): vol.In(
+                        keypad_button_names_to_leap.get(keypad_id)
+                    ),
+                }
+            ),
+        )
+
+    return keypad_trigger_schemas
+
+
+def _setup_lutronkeypad(
```

Comment:
```suggestion
@callback
def _async_setup_lutron_keypad(
```

For snake case, please separate words with `_`